src/Microsoft.DocAsCode.MarkdownLite/JsonTokenTreeRenderer.cs

```diff
@@ -0,0 +1,282 @@
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownStrongInlineToken token, MarkdownInlineContext context)
+        {
+            var childContent = StringBuffer.Empty;
+            foreach (var item in token.Content)
+            {
+                childContent += renderer.Render(item);
+            }
+            return this.Insert("Strong", childContent, Type.NonLeaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownEmInlineToken token, MarkdownInlineContext context)
+        {
+            var childContent = StringBuffer.Empty;
+            foreach (var item in token.Content)
+            {
+                childContent += renderer.Render(item);
+            }
+            return this.Insert("Em", childContent, Type.NonLeaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownCodeInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("InLineCode", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, GfmDelInlineToken token, MarkdownInlineContext context)
+        {
+            var childContent = StringBuffer.Empty;
+            foreach (var item in token.Content)
+            {
+                childContent += renderer.Render(item);
+            }
+            return this.Insert("Del", childContent, Type.NonLeaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTagInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("Tag", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownBrInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("Br", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTextToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("InLineText", this.GetSize(), Type.Leaf);
+        }
+
+        #endregion
+
+        #region Misc
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownIgnoreToken token, IMarkdownContext context)
+        {
+            return this.Insert("Ignore", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownRawToken token, IMarkdownContext context)
+        {
+            return this.Insert($"Raw(From{token.Rule.Name})", GetSize(), Type.Leaf);
+        }
+
+        #endregion
+
+        #region Private
+
+        private enum Type
+        {
+            Leaf,
+            NonLeaf
+        }
+
+        private StringBuffer Insert(StringBuffer name, StringBuffer tokenContent, Type tokenType)
+        {
+            StringBuffer result = "{\"name\":\"";
```

Comment:
But the StringBuffer is designed to deal with the string appending. use `"{\"name\":\"{name}\""`  will be Inefficient?

---
In reply to: [75074287](https://github.com/dotnet/docfx/pull/608#discussion_r75074287) [](ancestors = 75074287)